pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/store/StarTreeIndexMapUtils.java

```diff
@@ -198,9 +199,9 @@ public static List<Map<IndexKey, IndexValue>> loadFromInputStream(InputStream in
           column = StringUtils.join(split, KEY_SEPARATOR, 1, columnSplitEndIndex);
         }
         // Convert metric (function-column pair) to stored name for backward-compatibility
-        if (column.contains(AggregationFunctionColumnPair.DELIMITER)) {
-          AggregationFunctionColumnPair functionColumnPair = AggregationFunctionColumnPair.fromColumnName(column);
-          column = AggregationFunctionColumnPair.resolveToStoredType(functionColumnPair).toColumnName();
+        Optional<AggregationFunctionColumnPair> functionColumnPair = AggregationFunctionColumnPair.accept(column);
```

Comment:
Suggest doing a try-catch around `AggregationFunctionColumnPair.fromColumnName(column)` and simply ignore the exception as this pieces of code is purely for backward compatibility and is not regular code path. We should also add some comments
```suggestion
        if (column.contains(AggregationFunctionColumnPair.DELIMITER)) {
          try {
            AggregationFunctionColumnPair functionColumnPair = AggregationFunctionColumnPair.fromColumnName(column);
            column = AggregationFunctionColumnPair.resolveToStoredType(functionColumnPair).toColumnName();
          } catch (Exception e) {
            // Ignoring this exception for columns that are not metric (function-column pair)
          }
        }
```